### ISSUE TITLE
Let the End User and SImpliSafe have control still...

### ISSUE THREAD

**Issue by graanco** [nzapponi/homebridge-simplisafe3#35]
please change this line...
['-i', `https://${this.serverIpAddress}/v1/${this.cameraConfig.uuid}/flv?x=${width}`]
in the simlicam source
to ['-i', `https://${this.serverIpAddress}/v1/${this.cameraConfig.uuid}/flv`]

This way the video feed size is control by the settings from SimpliSafe Restful API and the user camera settings.

I would put in a pull request but it is so small of a thing that is would be more work.

**Comment by nzapponi** [comment]
The reason why Homebridge is defining the width and height is that HomeKit requests different definitions depending on which device is requesting to view the camera. For instance, an Apple Watch won't ever ask for 720p, and probably not even 480p.

That's why we're requesting a specific width to SimpliSafe, so that we get a scaled down feed already.. If you set your camera to 480p and Homebridge requests 720p, my understanding is that SimpliSafe should return a 480p feed.

If that's not the case, though, we can limit the max supported resolution within the plugin by reading SimpliSafe's API, but it'll be more complex than just changing this one line.

I'll add it as a potential enhancement to work on in the future.

**Comment by graanco** [comment]
Okay the way I see it and use it is the camsource gets the video scaling request from HK... SimpliSafe transmit the video scaling from user settings.  ffmpeg gets the video transcodes it to the needed scaling for hk. I missed a lot on this to figure it out the process. Only issue I’ve seen so far with SS is SS Media API isn’t ip6 compatible. yet.

**Comment by virtualjad** [comment]
Speaking of 720p - i noticed there wasn't any 1080p support in simplicam.js (the doorbell supports 1080p). Also logs show 720p resolution even though the cam is set to 1080p in the app. Is that something you plan to add by any chance?

**Comment by nikonratm** [comment]
Regarding @graanco s question, I think @nzapponi is correct. When I set my indoor camera max resolution to 480p and request the stream with any size > 480p, the stream itself is still only 480p:
`[10/24/2019, 3:08:00 PM] [Home Alarm]     Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), **848x480** [SAR 160:159 DAR 16:9], 20 fps, 20 tbr, 1k tbn, 40 tbc`

So therefore I vote we should close this issue as the "enhancement" isnt necessary, it already works as one would hope.

1080p support is easy, I just changed the default resolution parameters and added full HD to video options, works great for doorbell and indoor cam (which has max res of 720). Framerate is still crappy but I guess that's a tradeoff and hardware-dependent, at least this way users can still control it via SS settings. I'll push a PR for this.

But in general, worth remembering the size of feed is determined by the app you're using to view the stream. E.g. on my (15" retina) MBP it still seems to request a 1280x720 feed (I guess the mac app is limited to this?), same with iPhone, on my iPad Pro however it does seem to request a full HD 1080p stream.
